src/suggestion.rs

```diff
@@ -135,6 +158,131 @@ impl<'s> fmt::Display for Suggestion<'s> {
+            let mut misspelled_word = format!(
+                "{}",
+                self.literal.sub(Range {
+                    start: marker_range_relative.start - 1,
+                    end: marker_range_relative.end
+                })
+            );
+            if marker_size > TOO_LONG_WORD {
+                range_start_word.start = marker_range_relative.start - 1;
+                range_start_word.end = range_start_word.start + DISPLAYED_LONG_WORD;
+
+                range_end_word.start = marker_range_relative
+                    .end
+                    .saturating_sub(DISPLAYED_LONG_WORD);
+                range_end_word.end = marker_range_relative.end;
+
+                misspelled_word = format!(
+                    "{}...{}",
+                    self.literal.sub(range_start_word),
+                    self.literal.sub(range_end_word)
+                );
+                marker_size = misspelled_word.chars().count();
+            };
+            // right context has enough info to fill the terminal
+            //  |-----misspelled_word-----|--------right_context---------|
+            //
+            // Attempt to fit the misspelled word in the beginning followed by info.
+            if range_right_context.len() >= terminal_size {
+                // Left range will not be used in this case
+                range_left_context.start = 0usize;
+                range_left_context.end = 0usize;
+
+                range_right_context.start = marker_range_relative.end;
+                range_right_context.end = marker_range_relative.end
+                    + (terminal_size
+                        .saturating_sub(misspelled_word.chars().count() + PADDING_END + 1));
+                offset = offset.saturating_sub(range_left_context.start) + PADDING_OFFSET_START;
+            }
+            // left context has enough info to fill the terminal
+            // |---------left_context---------|-----misspelled_word-----|
+            //
+            // Attempt to fit the misspelled word with left context info
+            else if range_left_context.len() > terminal_size {
+                range_left_context.start = marker_range_relative.start.saturating_sub(
+                    terminal_size.saturating_sub(misspelled_word.chars().count() + PADDING_END),
+                );
+                range_left_context.end = marker_range_relative.start - 1;
+                // Right range will not be used
+                range_right_context.start = 0usize;
+                range_right_context.end = 0usize;
+
+                offset = offset.saturating_sub(range_left_context.start + 1) + PADDING_OFFSET;
+            }
+            // information will be shown in both sides of the `misspelled_word`
+            // |--left_context--|----misspelled_word---|--right_context--|
+            //
+            // Attempt to fit the misspelled word in the middle with info in th left and right of it
+            else {
```

Comment:
could this not be used to cover all 3 cases and only differentiate the 3 cases around `writeln!`?